pkgs/development/python-modules/geopy/default.nix

```
@@ -1,31 +1,64 @@
 { stdenv
 , buildPythonPackage
 , fetchPypi
+, fetchFromGitHub
 , isPy27
+, isPy37
 , mock
 , tox
 , pylint
+, geographiclib
 }:
 
-buildPythonPackage rec {
+let
   pname = "geopy";
-  version = "1.20.0";
-  disabled = !isPy27;
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "9419bc90ee6231590c4ae7acf1cf126cefbd0736942da7a6a1436946e80830e2";
+  commonBuildInputs = [ mock tox pylint ];
+
+  geopy27 = rec {
+    version = "1.20.0";
+    disabled = !isPy27;
+
+    src = fetchPypi {
+      inherit pname version;
+      sha256 = "9419bc90ee6231590c4ae7acf1cf126cefbd0736942da7a6a1436946e80830e2";
+    };
+
+    buildInputs = commonBuildInputs;
+    propagatedBuildInputs = [];
   };
 
-  doCheck = false;  # too much
+  geopy37 = {
+    version = "2.0.0";
+    disabled = !isPy37;
 
-  buildInputs = [ mock tox pylint ];
+    src = fetchFromGitHub {
+      owner  = "geopy";
+      repo   = "geopy";
+      rev    = "531b7de6126838a3e69370227aa7f2086ba52b89"; # unstable v2
+      sha256 = "07l1pblzg3hb3dbvd9rq8x78ly5dv0zxbc5hwskqil0bhv5v1p39";
+    };
 
-  meta = with stdenv.lib; {
-    homepage = "https://github.com/geopy/geopy";
-    description = "Python Geocoding Toolbox";
-    license = licenses.mit;
-    broken = true;
+    buildInputs = commonBuildInputs ++ [ geographiclib ];
```

```suggestion
    buildInputs = commonBuildInputs;
```

Anything in `propagatedBuildInputs` doesn't need to be added to `buildInputs`.

---

Ok thanks